src/AddComment/index.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+
+import TextArea from '../TextArea';
+import Button from '../Button';
+
+import './style.css';
+
+class AddComment extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      displaySuccess: false,
+      displayError: false
+    };
+  }
+
+  onChange = value => {
+    this.setState({ value });
+  };
+
+  onSubmit = event => {
+    event.preventDefault();
+    const { value } = this.state;
+    const { onCommentAdd } = this.props;
+
+    onCommentAdd({
+      variables: { body: value }
+    })
+      .then(({ data }) => {
+        this.setState({ displaySuccess: true });
+      })
+      .catch(error => {
+        this.setState({ displayError: true, errorMessage: error.toString() });
+      });
+  };
+
+  renderMessage = () => {
+    const { displaySuccess, displayError, errorMessage } = this.state;
+    if (displaySuccess) {
+      return <div className="AddComment-message AddComment-message--success">Your comment has been posted</div>;
+    } else if (displayError) {
+      return <div className="AddComment-message AddComment-message--error">{errorMessage}</div>;
+    } else {
+      return null;
+    }
+  };
+
+  render() {
+    const { value } = this.state;
+    return (
+      <div>
+        {this.renderMessage()}
```

Comment:
Usually I extract these things as own components like <RequestMessage /> but keep them still in the same file. However, good to see different approaches! Your way you have direct access to the state object! 👍

Reply:
I used a method instead of a separate component because it was not needed for reusability, but I'm sure this a premature optimization. I've also seen static methods used, so it would look like `
```
static renderMessage = ({ displaySuccess, displayError, errorMessage }) => {
  ...
}

render() {
  const { value, displaySuccess, displayError, errorMessage } = this.state;
  return (
      <div>
        <AddComment.renderMessage displaySuccess={displaySuccess} displayError={displayError} errorMessage={errorMessage} /> 
...
```

Curious to hear your thoughts on this.

Reply:
You are totally right! From a technical point of view it makes no sense to extract it unless you would want to reuse it or test it as standalone component. I guess for me it is more of a stylistic preference to extract atomic problem solvers as their own functions :)

Reply:
I first saw it used by @kentcdodds. It gives you the stylistic benefit of code separation without creating another component.

Reply:
Woohoo!

Reply:
Though truth be told I'd actually prefer to extract that method to be a pure function which accepts what it needs from props/state as an object argument. That way it's really clear what that part of the JSX needs, and it would also be pretty trivial to convert it to a full blown render prop later if desired.

Reply:
@kentcdodds thanks for the input! Not that it makes a difference, but do you mean option #1 or #2 below?

```
function MessageBox({ messageState, errorMessage }) {
  return {
    success: <div className="AddComment-message AddComment-message--success">Your comment has been posted</div>,
    error: <div className="AddComment-message AddComment-message--error">{errorMessage}</div>,
    none: null
  }[messageState];
}
...

class AddComment extends Component {

...

const messageState = isSuccess ? 'success' : isError ? 'error' : 'none';
  return (
    <div>
      <MessageBox messageState={messageState} errorMessage={errorMessage} /> <-- Option #1
      {MessageBox({ messageState, errorMessage })} <-- Option #2
```

Reply:
Option 2, but I'd lower-case it like a normal function. No need to make it a component necessarily. But really it doesn't matter all that much to me. Pretty loosy-goosy here.

```javascript
function renderMessage({ displaySuccess, displayError, errorMessage }) {
    if (displaySuccess) {
      return <div className="AddComment-message AddComment-message--success">Your comment has been posted</div>;
    } else if (displayError) {
      return <div className="AddComment-message AddComment-message--error">{errorMessage}</div>;
    } else {
      return null;
    }
  }
```